### PR TITLE
[OpenSSL] remove explicit apple deployment target flag

### PR DESCRIPTION
**Pull request by SSE4**
remove explicit [tools.apple_deployment_target_flag](https://docs.conan.io/en/latest/reference/tools.html#tools-apple-deployment-target-flag)

two reasons for the change:
1. since https://github.com/conan-io/conan/pull/7862, `AutoToolsBuildEnvironment` calls [tools.apple_deployment_target_flag](https://docs.conan.io/en/latest/reference/tools.html#tools-apple-deployment-target-flag) on its own
2. https://github.com/conan-io/conan/pull/8263 adds `os.sdk` sub-setting, https://github.com/conan-io/conan/pull/8264 adds `os.subsystem` sub-setting, they need to be passed to the [tools.apple_deployment_target_flag](https://docs.conan.io/en/latest/reference/tools.html#tools-apple-deployment-target-flag)

NOTE: I don't like the `AutoToolsBuildEnvironment` interface, hopefully new build helper will be much better...
the problem is `env_build.flags` doesn't contain all the final flags (e.g. apple-specific), as they are computed on the later stage. final flags computed are fortunately available in `env_build.var_dict` (as list) and in `env_build.vars` (as string).

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/openssl/1.x.x/conanfile.py [hunk 1]
```diff
@@ -433,16 +433,6 @@ def adjust_path(path):
     def _get_env_build(self):
         if not self._env_build:
             self._env_build = AutoToolsBuildEnvironment(self)
-            if self.settings.compiler == "apple-clang":
-                # add flags only if not already specified, avoid breaking Catalyst which needs very special flags
-                flags = " ".join(self._env_build.flags)
-                if "-arch" not in flags:
-                    self._env_build.flags.append("-arch %s" % tools.to_apple_arch(self.settings.arch))
-                if "-isysroot" not in flags:
-                    self._env_build.flags.append("-isysroot %s" % tools.XCRun(self.settings).sdk_path)
```
Comment (SpaceIm) [inline recipes/openssl/1.x.x/conanfile.py:442]: should it be removed?

Comment (SSE4) [reply recipes/openssl/1.x.x/conanfile.py:442]: yes, they are already there: https://github.com/conan-io/conan/blob/10e7d5b1e534cc665fa4c403cb560e5de20364d6/conans/client/build/autotools_environment.py#L359

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by SpaceIm — APPROVED** [review]

**Comment by prince-chrismc** [comment]
CI bot failed to update GitHub API after the last build

**Comment by jgsogo** [comment]
I'm really concern about these errors. CI is ok, but doesn't manage to update status 😠 

image

Merging manually...

**Pull request merged**